src/helpers.cpp

```diff
@@ -246,6 +246,29 @@ namespace ada::helpers {
     if (url.query.has_value()) return;
     while (!url.path.empty() && url.path.back() == ' ') { url.path.resize(url.path.size()-1); }
   }
+
+  ada_really_inline size_t find_authority_delimiter(const bool& is_special, std::string_view view) noexcept {
+    size_t location = 0;
+    bool location_found{false};
+
+    if (is_special) {
+      for (; location < view.size(); ++location) {
+        if (view[location] == '@' || view[location] == '/' || view[location] == '?' || view[location] == '\\') {
```

Comment:
We might find a clever way of auto-vectorizing this

Reply:
We might, yes.

Reply:
Please see https://github.com/ada-url/ada/pull/232... You might be able to re-use the same idea I applied.